### ISSUE TITLE
Accessibility Tweaks for SWAG Cms Extensions (Scroll Navigation)

### ISSUE THREAD

**Issue by Naitsabesss** [shopware/platform#14439]
### User story

Dear Shopware Team,

Developer support guided me to the issue tracker for my request.
we are using the scroll navigation from the swagcmsextensions and noticed that there are no ally tweaks yet. Since we are using ver. 6.6.10 (cms extensions 4.7) i wanted to see if there is accessibility support in later versions. But i wasnt able to finde the cmsextension in this repo.
So my questions are:
- Is there ally support for the swagcmsextensions esp. scroll navigation in later versions than 4.7. and what SW version do we need for that
- If no will there be ally support for the swagcmsextensions ?

Thanks for your time

FYI the ally screening came up with these issues:
Here's the translation:

Buttons require aria-label https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-1-1a-alternativtexte-fuer-bedienelemente
-    Up arrow has no Aria-label
-    Down arrow has no Aria-label
-    "X" Close must be a button and needs an aria-label

In opened state
-    Please create anchor navigation as a list, since it's an enumeration of elements, so feel free to use ul with individual li elements or rolte list/listitem  https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-3-1b-html-strukturelemente-fuer-listen

Meaningful sequence https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-3-2-sinnvolle-reihenfolge
-   When opened, the overlay content should be accessible. Currently you can't navigate further in the screen reader, only via Tab

Zoom & content loss
-    At 1280x769 and 200% zoom, content is slightly cut off (no content loss yet with 4 items) https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-4-4-text-auf-200-vergroesserbar
-    At 1290x1024px and 400% zoom, only 2 anchor points are visible and reachable via scroll, the rest is cut off (content loss) https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-4-10-inhalte-brechen-um

States & status messages https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-4-1-2-name-rolle-wert-verfuegbar
-    Aria-expanded=true + false > for status whether expanded or collapsed. Logic is comparable to an expanded and collapsed accordion
-    Set aria-current on the currently active link

User-defined settings https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-11-7-benutzerdefinierte-einstellungen
-    If I increase the font size in the browser, the anchor links are no longer visible?

Not a must, but anchor navigation would be nice as a landmark, i.e. section/region with aria-label="Contents of this page"

### Acceptance criteria

Cms Extensions contain ally tweaks

### Definition of Done

- [ ] Fulfills all acceptance criteria defined during discovery.
- [ ] Integration/E2E testing in staging is done.
- [ ] All integration/E2E/unit tests passing; all critical or high-priority bugs are resolved.
- [ ] Fulfills compliance, performance, security, and cloud-readiness needs.
- [ ] Observability—includes monitoring, alerting, and logging; incident response handbooks updated.
- [ ] Documentation—developer docs are written or updated, including information on how the feature or change is adopted and tested. If new functionality, end user documentation is written or updated.
- [ ] (conditional) All new API endpoints have a defined API schema.
- [ ] (conditional) If the change affects the Storefront, it has been tested in the latest three versions of Firefox, Chrome, Edge, IE11, and Safari, as well as the different viewports and devices (Desktop / Tablet / Mobile).
- [ ] (conditional) If the change affects the Administration, it has been tested in current versions of Firefox, Chrome, and Edge.
- [ ] (conditional) Functionality that includes new API routes are callable via the Store-API (/store-api/[version]/).
- [ ] (conditional) If a new Admin module is created, the access control layer is provided, and permissions can be managed via the Users & Permissions module.
- [ ] (conditional) If a new UI input field or interactive element is introduced, the ACL wrapper is added.
- [ ] (conditional) Performance testing done if the change has a high performance impact, such as affecting scalability, load balancing, or query optimization.

**Comment by ravenokko** [comment]
Hi @Naitsabesss and thank you for reaching out! We'll take a look and get this fixed.

**Comment by ravenokko** [comment]
Thank you for your patience, this took a bit of time to untangle. Accessibility improvements have been added starting in version 4.7 of the CMS-Extensions. For shops on 6.6.x, you will need the [ACCESSIBILITY_TWEAKS flag](https://developer.shopware.com/docs/guides/development/accessibility/storefront-accessibility.html#how-to-deal-with-breaking-accessibility-changes) enabled for them to appear. Some of these issues have already been addressed, and I have created sub issues for the remaining points that are valid.

I will try to address each of your individual points:

> Buttons require aria-label https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-1-1a-alternativtexte-fuer-bedienelemente
> 
> Up arrow has no Aria-label
> 
> Down arrow has no Aria-label
> 
> "X" Close must be a button and needs an aria-label

This has already been addressed.


> In opened state
> 
> Please create anchor navigation as a list, since it's an enumeration of elements, so feel free to use ul with individual li elements or rolte list/listitem https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-3-1b-html-strukturelemente-fuer-listen

I'm assuming here you specifically mean the scroll navigation, which follows this pattern.

> Meaningful sequence https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-3-2-sinnvolle-reihenfolge
> 
> When opened, the overlay content should be accessible. Currently you can't navigate further in the screen reader, only via Tab

Screen readers should be able to access all content in the Quickview overlay and the Scroll navigation


>Zoom & content loss
>
>At 1280x768 and 200% zoom, content is slightly cut off (no content loss yet with 4 items) https://bitvtest.de/pruefschritt/bitv->20-web/bitv-20-web-9-1-4-4-text-auf-200-vergroesserbar
>
>At 1290x1024px and 400% zoom, only 2 anchor points are visible and reachable via scroll, the rest is cut off (content loss) >https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-1-4-10-inhalte-brechen-um

Some content has been adjusted for smaller viewports per responsive design, but there were still some reflow/resizing issues which will be addressed in https://github.com/shopware/shopware/issues/16030. However, this solution will also partially depend on separate fixes in platform which needs to be addressed in a separate issue.


> States & status messages https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-9-4-1-2-name-rolle-wert-verfuegbar
> 
> Aria-expanded=true + false > for status whether expanded or collapsed. Logic is comparable to an expanded and collapsed accordion

I'm not quite sure where this applies, as the Quickview is technically closer to a modal. Can you give more information on where exactly this applies?


> Set aria-current on the currently active link 

This has been added to a separate issue https://github.com/shopware/shopware/issues/16028 which has been added to the backlog



> User-defined settings https://bitvtest.de/pruefschritt/bitv-20-web/bitv-20-web-11-7-benutzerdefinierte-einstellungen
> 
> If I increase the font size in the browser, the anchor links are no longer visible?

Will be addressed in https://github.com/shopware/shopware/issues/16029
